tests/component-tests/textToSpeech/js/TextToSpeechTests.js

```diff
@@ -224,38 +242,50 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
         jqUnit.assertFalse("Shouldn't be paused", that.model.paused);
     };
 
-    fluid.tests.textToSpeech.bypassTest = function (bypassMessage) {
-        jqUnit.test("Tests were skipped.", function () {
-            jqUnit.assert(bypassMessage);
-        });
+    fluid.tests.textToSpeech.baseTests = function () {
+        fluid.test.conditionalTestUtils.chooseTestByPromiseResult("Confirming if TTS is available for initialization and start/stop tests", fluid.textToSpeech.checkTTSSupport, fluid.tests.textToSpeech.ttsTestEnvironment, fluid.test.conditionalTestUtils.bypassTest, "Browser appears to support TTS", "Browser does not appear to support TTS");
+    };
+
+    fluid.tests.textToSpeech.notLinuxTests = function () {
+        fluid.test.conditionalTestUtils.chooseTestByPromiseResult("Confirming if TTS is available for pause and resume tests", fluid.textToSpeech.checkTTSSupport, fluid.tests.textToSpeech.ttsPauseResumeTestEnvironment, fluid.test.conditionalTestUtils.bypassTest, "Browser appears to support TTS", "Browser does not appear to support TTS");
     };
 
-    // Chooses which test function to execute based on the results of a
-    // promise; wraps the promise in an asyncTest to cause QUnit's test
-    // runner to suspend while the decision is being made asynchronously by
-    // the promise. Without this, QUnit will merrily proceed along to the
-    // next test set, which can cause errors various contexts including the
-    // all-tests runner.
-    //
-    // wrapperMessage, task, resolveFunc, rejectFunc: required
-    // "task" must be a function returning a promise
-    // resolveMessage, rejectMessage: optional strings, passed to the test
-    // functions
-    fluid.tests.textToSpeech.chooseTestByPromiseResult = function (wrapperMessage, task, resolveFunc, rejectFunc, resolveMessage, rejectMessage) {
-        resolveMessage = resolveMessage || "Promise resolved, running resolve test.";
-        rejectMessage = rejectMessage || "Promise rejected, running reject test.";
-        jqUnit.asyncTest(wrapperMessage, function () {
-            jqUnit.expect(0);
-            task().then(function () {
-                jqUnit.start();
-                resolveFunc(resolveMessage);
-            }, function () {
-                jqUnit.start();
-                rejectFunc(rejectMessage);
-            });
+    fluid.tests.textToSpeech.runTests = function (that) {
```

Comment:
This is generic and can go into your testUtils - factor off a grade from fluid.tests.textToSpeech.testRunner to account for the fact that it has certain material in the option named "tests" and document its structure